sympy/functions/special/gamma_functions.py

```diff
@@ -943,7 +943,11 @@ def _eval_rewrite_as_intractable(self, z, **kwargs):
         return log(gamma(z))
 
     def _eval_is_real(self):
-        return self.args[0].is_real
+        z = self.args[0]
+        if z.is_positive:
+            return True
+        elif z.is_real:
+            return False
```

Comment:
If `z.is_positive` is None but `z.is_real` is True then this will return False. However in some cases it should return None e.g. `z = Symbol('z', real=True)`. So maybe this test should be `elif z.is_nonpositive`.